arch/risc-v/src/common/riscv_internal.h

```diff
@@ -232,6 +232,12 @@ void riscv_pminitialize(void);
 #  define riscv_pminitialize()
 #endif
 
+/* DMA **********************************************************************/
+
+#ifdef CONFIG_ARCH_DMA
+void weak_function riscv_dma_initialize(void);
```

Comment:
Maybe better change to macro if DMA is not supported? Or we have a case when DMA is supported, but not inited?

Reply:
Some drivers call init inside of the specific controller driver  using the DMA (I2C, SPI, etc...), so we cannot purely condition calling this function based on the config definition for DMA support. This is why I brought over the approach we use for ARM.